### PR TITLE
Add switch-change attribute (closes #80)

### LINKED ISSUE #80: When a user clicks the switch an event or method should be able to be called.
# Question
How can we bind an event to a toggle?

# Story
As a developer I should be able to trigger a method call when a switch is clicked.

### PR DESCRIPTION
**Pull request by icholy**
Addresses #80 I wanted to keep it similar to the `ngChange` directive.

### CODE REVIEW

**Comment by icholy** [comment]
@frapontillo looks like there's something wrong with the travis setup.

File: test/spec/directives/bsSwitchSpec.js [hunk 1]
```diff
@@ -652,4 +656,25 @@ describe('Directive: bsSwitch', function () {
   it('should watch updates in getterSetter', inject(makeTestGetterSetter()));
   it('should watch updates in getterSetter', inject(makeTestGetterSetter(true)));
 
+  function makeTestChange(input) {
+    return function () {
+      compileDirective('change', input);
+
+      var invoked = false;
+      scope.change = function() {
```
Comment (frapontillo) [inline test/spec/directives/bsSwitchSpec.js]: You could actually create a spy for this (see jasmine spec)

File: src/directives/bsSwitch.js [hunk 2]
```diff
@@ -52,6 +52,20 @@ angular.module('frapontillo.bootstrap-switch')
...
 
         /**
+         * Returns a function that executes the provided expression
+         *
+         * @param value The string expression
+         * @return a function that evaluates the expression
+         */
+        var getExprFromString = function (value) {
+          if (angular.isUndefined(value)) {
+            return function () {};
```
Comment (frapontillo) [inline src/directives/bsSwitch.js]: Can you return `angular.noop`?

**Comment by frapontillo** [comment]
@icholy thank you for this!
Could you please remove the dist file from the PR and address the comments?
Also, if you are able to do it, please update the travis conf file so that it uses Node 4 LTS and Node 5? (This should solve the Travis issue we have right now)

**Comment by icholy** [comment]
@frapontillo according to [this](https://docs.travis-ci.com/user/languages/javascript-with-nodejs#Provided-Node.js-Versions) I don't think travis support 5.0 yet.

**Comment by frapontillo** [comment]
@icholy you are right, but we may require gcc (see https://docs.travis-ci.com/user/languages/javascript-with-nodejs#Node.js-v4-(or-io.js-v3)-compiler-requirements)

**Comment by icholy** [comment]
@frapontillo I removed the changes to travis and rebased/squashed.

**Comment by MtDalPizzol** [comment]
Hey @icholy , I missed the point of having a function to parse the expression. Maybe it's my ignorance on angular directives. Could you give a look at what I suggested at #80 ?

**Comment by icholy** [comment]
@MtDalPizzol I'm making it consistent with `ngChange`.

**Comment by MtDalPizzol** [comment]
@icholy , right... I'll take a look at that. Thanx.

Strange... I have the example I posted running here. =/

**Comment by icholy** [comment]
@MtDalPizzol You're doing a one way binding to the isolate scope using `&` and expecting that to be a function. But you're passing it an expression `switch-change="updateState(item)"`. I'd expect this not to work. You'd have to just pass it the function `switch-change="updateState"`. But in that case you don't have the option to pass it a parameter. That makes it kinda useless for the `ngRepeat` use case.

**Comment by MtDalPizzol** [comment]
Well...  I checked the code here, and it's working. Anyway, your solution looks more robust than mine. I'l try that later, as @frapontillo suggested.

**Comment by icholy** [comment]
@frapontillo you going to merge?

**Comment by frapontillo** [comment]
This has been merged into develop (73d798a414fb6dc54e9feb4f1abeb51021d4ddee) after updating all the npm packages and the Travis configuration, which now confirms that everything is working.



Thanks @icholy!

**Comment by ArthurPBressan** [comment]
@icholy's code isn't working.

Consider:

 ```html
    <span ng-controller="switchesCtrl" ng-cloak>
        {{ switchState }}
        {{ counter }}
        <input bs-switch ng-model="switchState" type="checkbox" 
        switch-change="switchChange()">
    </span>
```

```javascript
switchesControllers.controller('switchesCtrl', ['$scope',
  function($scope) {
    $scope.switchState = false;
    $scope.counter = 0;

    $scope.switchChange = function() {
        $scope.counter++;
    }
});
```

I would have expected the counter to increment each time I changed the switch state, regardless of previous state. In reality, the function isn't firing at all.

However, @MtDalPizzol's solution that he posted on #80 works as expected.

**Comment by frapontillo** [comment]
@ArthurPBressan can you share a plunker?

**Comment by tanzaho** [comment]
I quickly made a plunkr based on @ArthurPBressan code : https://plnkr.co/edit/PJ3HNXSeWe54LMybRyxT?p=preview

It only works after the second change.

(Can't get it to work on my project yet)

**Comment by tanzaho** [comment]
I slightly changed the Plunkr and included the compiled script instead of separately the module and directive : 
Here is the new [Plunkr](https://plnkr.co/edit/CuisVXHk9FIa4d95t4dy?p=preview).
By doing this, it doesn't work as initially reported by @ArthurPBressan .

I'm sorry I'm not an JS guy at all. My new Plunkr might be rather stupid. I just tried to understand the difference between my project and the first plunkr and discovered this.

**Comment by icholy** [comment]
@tanzaho the scripts in the dist folder haven't been updated yet. I looked at your plunker and I know what the problem is. If you put a `console.log` in, you'll see the `switchChange` is being called when it's supposed to be. The issue is that it's not in a digest cycle. I'll fix that.

**Comment by ArthurPBressan** [comment]
Is there any way you can modify the switchChange to pass in an argument to the event function? Or is that something that should work already?

**Comment by icholy** [comment]
@ArthurPBressan `switch-change` takes an expression. So yes.

@frapontillo I'm going to wrap the change callback in a `$timeout` but I need to update the tests. I'll open another PR when it's ready.

**Comment by tanzaho** [comment]
Thanks @icholy. Sorry for the useless comment earlier : I should have seen that myself from the modification date of the file. 
Great to know you're working on a fix, good job.
I'll be there to test :-)

**Comment by icholy** [comment]
@tanzaho no worries, that plunker was a big help.

**Pull request closed without merging**